src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -585,6 +578,11 @@ protected virtual Task<bool> HandleSignOutCallbackAsync()
                         tokenEndpointResponse = await RedeemAuthorizationCodeAsync(tokenEndpointRequest);
                     }
 
+                    if (!string.IsNullOrEmpty(tokenEndpointResponse.Error))
```

Comment:
Addressed, see the latest commit.

Overall, the focus here is that if authentication response status is 200 but with an `error` in body, do we consider it an error and interrupt the operation. The change I put in the second commit will allow the authentication to move one in this situation.

In real world, I doubt if there is any authentication provider will do this. (How to set status code for failed authentication response is not defined in the spec.)